src/Detector/Runtime.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Utopia\Detector\Detector;
+
+use Utopia\Detector\Detection\Runtime as DetectionRuntime;
+use Utopia\Detector\Detector;
+
+class Strategy
+{
+    public const FILEMATCH = 'filematch';
+
+    public const EXTENSION = 'extension';
+
+    public const LANGUAGES = 'languages';
+
+    private string $value;
+
+    public function __construct(string $value)
+    {
+        if (! in_array($value, [self::FILEMATCH, self::EXTENSION, self::LANGUAGES])) {
+            throw new \InvalidArgumentException("Invalid strategy: {$value}");
+        }
+        $this->value = $value;
+    }
+
+    public function getValue(): string
+    {
+        return $this->value;
+    }
+}
+
+class Runtime extends Detector
+{
+    public function __construct(
+        protected array $inputs,
+        protected Strategy $strategy,
+        protected string $packager = 'npm'
+    ) {}
+
+    public function detect(): ?DetectionRuntime
+    {
+        switch ($this->strategy->getValue()) {
+            case Strategy::FILEMATCH:
+                foreach ($this->options as $detector) {
+                    $detectorFiles = $detector->getFiles();
+
+                    $matches = \array_intersect($detectorFiles, $this->inputs);
+                    if (\count($matches) > 0) {
```

Comment:
Same comment as before regarding count vs nested-foreach; let's find out whats best strategy, and do that consistently everywhere